imt/node.go

```diff
@@ -1,21 +1,39 @@
 package imt
 
 import (
+	"encoding/binary"
 	"errors"
 	"math/big"
 )
 
 type Node struct {
 	Key     *big.Int
```

Comment:
What's the reason to keep storing the `Key` in the node? I cloned, removed it locally and made the changes so that everything compile and I did not notice any issue.

Reply:
This is simply to make the hash function simpler. The alternative might be to pass the key directly to the hash function.

We could instead hash the index instead of the key; I believe that would work. But it would require us to change the circuits.